lldb/source/API/SBDebugger.cpp

```diff
@@ -226,13 +227,45 @@ lldb::SBError SBDebugger::InitializeWithErrorHandling() {
   return error;
 }
 
+#if LLVM_ENABLE_TELEMETRY
+#if ENABLE_BACKTRACES
+static void TelemetryHandler(void *) {
+  // TODO: get the bt into the msg?
+  // Also need to pre-alloc the memory for this entry?
+  lldb_private::telemetry::DebuggerInfo entry;
+  entry.exit_desc = {-1, ""};
+  if (auto* instance = lldb_private::telemetry::TelemeryManager::getInstance()) {
+    if (instance->GetConfig()->EnableTelemetry()) {
+      instance->AtDebuggerExit(&entry);
+    }
+  }
+}
+
+static bool RegisterTelemetryHander() {
+  sys::AddSignalHandler(TelemetryHandler, nullptr);
+  return false;
+}
+#endif
+#endif
+
+static void InstallCrashTelemetryReporter() {
+#if LLVM_ENABLE_TELEMETRY
+
```

Comment:
@labath Is this the right place to add the crash-handler?

Reply:
Maybe? I don't think there's a single right place to do that, as each option comes with some tradeoffs. It may be best to leave this all up to downstream..

Reply:
agreed.